sidecar/dns/server.go

```diff
@@ -214,38 +347,39 @@ func (s *Server) findMatchingServices(question dns.Question, request, response *
 
 }
 
-func validateEndPointTypeTCPAndUDP(value string) (net.IP, error) {
-	ip, _, err := net.SplitHostPort(value)
+func validateEndPointTypeTCPAndUDP(value string) (net.IP,string, error) {
+	ip, port, err := net.SplitHostPort(value)
 	if err != nil {
-		return nil, err
+		return nil,port, err
 	}
 	parsedIP := net.ParseIP(ip)
 	if parsedIP == nil {
-		return nil, fmt.Errorf("tcp/udp ip value %s is not a valid ip format", ip)
+		return nil,port, fmt.Errorf("tcp/udp ip value %s is not a valid ip format", ip)
 	}
-	return parsedIP, nil
+	return parsedIP,port, nil
 }
 
-func validateEndPointTypeHTTP(value string) (net.IP, error) {
-	startsWithHttp := strings.HasPrefix(value, "http://")
-	startsWithHttps := strings.HasPrefix(value, "https://")
-	if !startsWithHttps && !startsWithHttp {
+func validateEndPointTypeHTTP(value string) (net.IP,string, error) {
+	startsWithHTTP := strings.HasPrefix(value, "http://")
+	startsWithHTTPS := strings.HasPrefix(value, "https://")
+	if !startsWithHTTPS && !startsWithHTTP {
 		value += "http://"
```

Comment:
This appends "http://" to the end of value, e.g. `172.107.0.10:8080http://`. Not what you want :)